content/projects/gsoc/2020/projects/git-plugin-performance.adoc

```diff
@@ -23,4 +22,41 @@ links:
   idea: /projects/gsoc/2020/project-ideas/git-plugin-performance
 ---
 
-// TODO: add details here, remove `showGoogleDoc: true`
+The project aims to improve overall performance of git plugins by applying micro-benchmarking principles to create a comparitive study of CLI and JGit implementations of "git" in the plugins. The comparitive study can be used to codify the optimal selection of git implementation for an operation. Furthermore, it aims to fix non-trivial issues related to redundancy at hot code paths in git plugin.
+
+=== Project Deliverables
+* Integration of JMH: micro-benchmark module addition to test module of git plugins
+  ** This includes both utilities to support the creation of a performance benchamrking test environment in git-plugin and benchmark tests written for git operations critical to the plugin's end-to-end performance
+* A new additional behavior in GitSCM checkout in build configuration
+  ** The behavior is supposed to enable performance, which will in turn selectively prioritize CliGitAPIImpl or JGitAPIImpl for the best performance
+* A working fix for link:https://github.com/jenkinsci/git-plugin/pull/845[git double/redundant fetch issue]
+  ** A fix which ensures no repository/user data loss if second fetch is removed with automated tests to confirm the intended behaviour
+* Comparison of git clone and git fetch performance
+  ** Implementation of git clone with credentials support without breaking compatibility
```

Comment:
Possible additional item?
```suggestion
  ** Implementation of git clone with credentials support without breaking compatibility
  ** Clear identification of plugin features that cannot be implemented with `git clone`
```

Reply:
https://issues.jenkins-ci.org/browse/JENKINS-23345 and https://issues.jenkins-ci.org/browse/JENKINS-30318 are great discussions on why git (init + fetch) is used instead of git clone and how implementing git clone might break existing use-cases provided by git plugin. 

I'll be adding this in the design document as well.

Reply:
@MarkEWaite Yes, I think it would be a great way to not break compatibility with the existing use cases, I'll add this item in the design document!
Thanks.